src/nvim/fileio.c

```diff
@@ -1883,16 +1883,20 @@ readfile (
       xfree(keep_msg);
       keep_msg = NULL;
       msg_scrolled_ign = TRUE;
-      p = msg_trunc_attr(IObuff, FALSE, 0);
+
+      if (!read_stdin && !read_buffer) {
+        p = msg_trunc_attr(IObuff, FALSE, 0);
```

Comment:
Oh, yes! While writing this, I set `p = NULL` at the beginning of the function without checking where it is actually used and I thought that would do it. Lession learned.

I will write a small PR for this as soon as I can.